lib/libzfs/libzfs_dataset.c

```diff
@@ -2120,6 +2121,11 @@ get_numeric_property(zfs_handle_t *zhp, zfs_prop_t prop, zprop_source_t *src,
 
 	*source = NULL;
 
+	if ((prop == ZFS_PROP_CREATETXG) &&
+	    ((*val = zhp->zfs_dmustats.dds_creation_txg) > 0)) {
+		return (0);
+	}
+
```

Comment:
Why it is not in underlying switch(prop) among other properties?

Reply:
The original request was to add createtxg support for fast snapshot iteration, which was supported by name property previously. Since we ensure that this property is available for both simple and normal dataset handlers, we directly return the value here instead of parsing nvlist. Moreover, we do not populate nvlist for fast path, instead just set the createtxg.

Reply:
Look for case ZFS_PROP_NUMCLONES: below.  What is the difference?

Reply:
Thank you for pointing this out. Yes, totally agreed with you. Updated.